.github/workflows/validate.js

```diff
@@ -0,0 +1,102 @@
+const {spawnSync} = require('child_process');
+const parseDiff = require('parse-diff');
+
+const baseRef = process.env.GITHUB_BASE_REF;
+
+console.error('base ref:', baseRef);
+
+const proc = spawnSync('git', [
+	'diff',
+	`origin/${baseRef}..HEAD`,
+	'--oneline'
+]);
+
+if (proc.error) throw proc.error;
+
+if (proc.status !== 0) {
+	console.error('There was a problem fetching the diff:');
+	console.error(proc.stderr.toString('utf-8'));
+	process.exit(1);
+}
+
+const diffOutput = proc.stdout.toString('utf-8');
+const diff = parseDiff(diffOutput);
+
+console.error('\n\n############## DIFF ##############');
+console.error(diffOutput);
+console.error('############## END DIFF ##############\n\n');
+
+const readme = diff.filter(({from, to}) => (from === to) && (from == 'README.md'));
+
+if (readme.length !== 1) {
+	console.error('PR changes files other than README');
+	process.exit(1);
+}
+
+const chunks = readme[0].chunks;
+
+if (chunks.length !== 1) {
+	console.error('Too many things changed in the README.');
+	process.exit(1);
+}
+
+let addedIndex = -1;
+const added = chunks[0].changes.filter((c, i) => {
+	if (c.add) {
+		addedIndex = i;
+		return true;
+	}
+
+	return false;
+});
+
+if (added.length !== 1) {
+	console.error('Expected exactly one added line');
+	process.exit(1);
+}
+
+if (addedIndex === 0) {
+	console.error('Added line to beginning of README.');
+	process.exit(1);
+}
+
+if (addedIndex === chunks[0].changes.length - 1) {
+	console.error('Added line to end of README (not able to automatically check)');
+	process.exit(1);
+}
+
+const linePattern = /^\- ([^,]+), @([^ ,]+)(?: \([^)]+\))?$/;
```

Comment:
is this checking for alphabetization by first name? we're doing it by last name. noticed it failing consistently

Reply:
Ah, yes.